### PR TITLE
Adding dark theme

### PR DESCRIPTION
**Pull request by Viniciud**
This pull request adds the necessary configs for the dark mode and a new module called theme, which configures the theme on the application.

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -44,7 +54,6 @@
               "optimization": true,
               "outputHashing": "all",
               "sourceMap": false,
-              "extractCss": true,
```
Comment (Savjee) [inline angular.json:47]: Why was this option removed?

File: src/app/theme/services/theme.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+@Injectable({
+  providedIn: "root",
+})
+export class ThemeService {
+
+  public static default = 'light';
+
+  public get current(): string {
+    return localStorage.getItem('theme') ?? ThemeService.default;
```
Comment (Savjee) [inline src/app/theme/services/theme.service.ts:11]: I understand that you allow users to manually switch between light/dark mode.
Would be nice if you could respect the browser's appearance first.

Would be great if you could add this.
If not, I can work on this later.

File: src/app/theme/services/theme.service.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    localStorage.setItem('theme', value);
+    this.style.href = `${value}.css`;
+  }
+
+  private readonly style: HTMLLinkElement;
+
+  constructor() {
+    this.style = document.createElement('link');
+    this.style.rel = 'stylesheet';
+    document.head.appendChild(this.style);
```
Comment (Savjee) [inline src/app/theme/services/theme.service.ts:24]: Not really a fan of dynamically loading the CSS when needed.
Could you rework this a bit and have all CSS styles served directly to the user?

You can use a CSS selector to detect if the browser is using darkmode:
```css
@media (prefers-color-scheme: dark) {
    /* Rules here */
}
```

Or, you could set a class on the body to indicate the user has switched to dark mode.
Eg: `<body class="dark-mode">`

File: src/light.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@import '~bootstrap/scss/bootstrap';
+@import "~bootstrap-icons/font/bootstrap-icons.css";
+
+body, * {
+  margin: 0;
+  border: 0;
+  box-sizing: border-box;
+}
```
Comment (Savjee) [inline src/light.scss:8]: I feel like we don't need a "light.css" file at all.
Couldn't this be put inside the regular CSS file?

**Review by Savjee — CHANGES_REQUESTED** [review]
Thanks a lot for this PR!
Some small remarks to think about.

**Comment by Viniciud** [comment]
Thank you for that opportunity. I will work on that changes and send them again.

**Comment by Viniciud** [comment]
In the last change I made, the app can detect the user theme preferences

File: src/light.scss (hunk 4 above)
Comment (Viniciud) [reply src/light.scss]: For sure, in the last change I did that.

**Review by Savjee — APPROVED** [review]
Awesome! Thanks a lot.

**Comment by Viniciud** [comment]
I'm glad to help and congrats again on the great work!

**Pull request merged**